### PR TITLE
feature/ Past Groups

### LINKED ISSUE #51: API Call to return all roles that an user has had over the years
REST Call: /users/{user_id}/past_groups

Needed for <https://github.com/makeadiff/Madnet/issues/24>

### Sub tasks...

- [ ] Swagger documentation - api/swagger/swagger.yaml
- [ ] REST API Call - routes/api.php
- [ ] GraphQL Version of the call - graphql-schema/schema.graphql
- [ ] Tests - tests/Unit/GroupTest.php

### PR DESCRIPTION
**Pull request by Bhanu-mbvg**
Fixes #51 

- [ ] REST API -> /users/1/past_groups
- [ ] Graphql Version  -> { user(id: 1) { past_groups { id name year }} }

### CODE REVIEW

File: api/swagger/swagger.yaml [hunk 1]
```diff
@@ -386,6 +386,25 @@ paths:
...
+      tags:
+        - user
+        - group
+      summary: Past User Group of an user
+      description: Return all the groups both present and past of the given user ID.
+      parameters:
+        - name : user_id
+          type: integer
+          in: path
+          required: given_to_user_name
```
Comment (binnyva) [inline api/swagger/swagger.yaml:400]: This is a `true` / `false`

File: graphql-schema/schema.graphql [hunk 2]
```diff
@@ -214,6 +215,14 @@ type Group {
     users: [User]
 }
 
+type PastGroups {
```
Comment (binnyva) [inline graphql-schema/schema.graphql:218]: I am thinking a better way of doing this is just adding a `year` field to the `Group` type. Adding a new type might not be required.

File: graphql-schema/schema.graphql [hunk 3]
```diff
@@ -127,6 +127,7 @@ type User {
     sex: String
     status: String
     groups: [Group]
+    past_groups: [PastGroups] @hasMany(relation: "pastGroups")
```
Comment (binnyva) [inline graphql-schema/schema.graphql:130]: `past_groups: [Group]  @hasMany(relation: "pastGroups")`
Might be a better/simpler option.

**Review by binnyva — CHANGES_REQUESTED** [review]
This is pretty good. Some minor changes requested :-D

**Comment by Bhanu-mbvg** [comment]
@Bhanu-mbvg made the requested changes

**Pull request merged**